CONTRIBUTING.md

```diff
@@ -1,88 +1,203 @@
-
-## General rules
-
-* DO NOT require that users perform any extensive initialization before they can start programming basic scenarios.
-* DO NOT use regions. DO use partial classes instead.
-* DO NOT seal controls.
-* DO NOT use verbs that are not already used like fetch.
-* DO NOT return true or false to give sucess status. Throw exceptions if there was a failure.
-* DO provide good defaults for all values associated with parameters, options, etc.
-* DO ensure that APIs are intuitive and can be successfully used in basic scenarios without referring to the reference documentation.
-* DO communicate incorrect usage of APIs as soon as possible.
-* DO design an API by writing code samples for the main scenarios. Only then, you define the object model that supports those code samples.
-* DO declare static dependency properties at the top of their file.
-* DO use extension methods over static methods where possible.
-* DO use verbs like GET.
-
-## Naming conventions
-
-* We are following the coding guidelines of [.NET Core coding style](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md).
-
-## Documentation
-
-* DO NOT expect that your code is so well designed that it needs no documentation. No code is that intuitive.
-* DO provide great documentation with all new features and code.
-* DO use readable and self-documenting identifier names.
-* DO use consistent naming and terminology.
-* DO provide strongly typed APIs.
-* DO use verbose identifier names.
-
-## Files and folders
-
-* DO associate no more than one class per file.
-* DO use folders to group classes based on features.
+You can contribute to this project by contributing to:
+
+* [Issues](https://github.com/microsoft/TemplateStudio/issues)
+* [Discussions](https://github.com/microsoft/TemplateStudio/discussions)
+* [Templates](#Templates)
+
+If you intend to contribute code changes, learn how to [set up your development environment](#Set-up-your-development-environment).
+
+When contributing template changes, [validate](#Validating-changes) your changes by generating projects with updated templates, then file a PR to trigger CI validation.
+
+This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
+For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or
+contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
+
+## Set up your development environment
+
+* Install [Visual Studio 2022 (.NET Desktop and Visual Studio Extension Workloads)](https://visualstudio.microsoft.com/thank-you-downloading-visual-studio/?sku=Community&channel=Release&version=VS2022&source=VSLandingPage&passive=false)
+* Install [Visual Studio Code](https://code.visualstudio.com/Download)
```

Comment:
Why is installing VSCode necessary?
Even mentioning it may cause the reader to think that TS works with VSCode.

Reply:
I prefer it when searching through the repo, editing files like pipelines that aren't part of any solution, updating md files, etc.

Reply:
> I prefer it when searching through the repo, editing files like pipelines that aren't part of any solution, updating md files, etc.

Then an explanatory note would be useful. The instructions here say to install it but it's not called out where/when/why to use it.

Reply:
Added a note.